include/extractor/guidance/intersection_generator.hpp

```diff
@@ -90,8 +90,8 @@ class IntersectionGenerator
     // The treatment results in a straight turn angle of 180º rather than a turn angle of approx
     // 160
     OSRM_ATTR_WARN_UNUSED
-    Intersection MergeSegregatedRoads(const NodeID intersection_node,
-                                      Intersection intersection) const;
+    ConnectedRoads MergeSegregatedRoads(const NodeID intersection_node,
+                                        ConnectedRoads intersection) const;
```

Comment:
if it's important to have it named `ConnectedRoads` to you, then we should also change the name `intersection` into `connected_roads` to not loose sync between type and name. Right now (`Intersection intersection`) connects the variable to its type